### DISCUSSION TITLE
Connection to Relay Server from external to internal Network fails

### DISCUSSION THREAD

**Discussion by DerArchitekt17** [rustdesk/rustdesk#7574] [Q&A] [answered]
Hello everyone,

I'm really at the end of my best knowledge and research, therefore I'd appreciate every help possible in this affair.

I have set up a RustDesk server (V 1.1.10-3) within a network (10.0.0.3/24), alongside with a Windows Server (10.0.0.2/24) which acts among other roles as DNS Server and an OPNsense firewall, that acts as a router (internally 10.0.0.1/24, externally 1.1.1.1). RustDesk runs on Ubuntu 22.04.4 LTS Server via by a script called command "./hbbs 10.0.0.3" managed by systemctl. The exact command RustDesk Server runs on is "./hbbs 10.0.0.3".

The firewall forwards traffic, that matches the ports 21115 - 21117 TCP / UDP, directly to the RustDesk Server. The RustDesk server's internal firewall also permits traffic from / to the ports 21115 - 21117 TCP / UDP.

There is also a public DNS record (let's say "rustdesk.contoso.com"), that points to the firewall's public address (1.1.1.1).

The RustDesk client applications are all configured with the "rustdesk.contoso.com" domain as ID server, with the corresponding key.

So far, all RustDesk connections between external clients (clients that are not within the same network as the RustDesk server) are working absolutely fine. However, as soon as I want to connect to the Windows Server from an external RustDesk Client and vice versa, there is no connection being established. The particular error is "Failed to connect via relay server", shown by whoever initiates the connection.

What I've done so far for on the Windows Server, I'd like to connect to (all with a negative result):
- Set the ID Server value to the local IP address (10.0.0.3)
- Set the ID Server value to the public Firewall IP address (1.1.1.1)
- Set the ID server to the public FQDN (rustdesk.contoso.com)
- Create a DNS record on the Windows Server that points from the FQDN to the local IP address, then set the ID server parameter to the FQDN
- Create a DNS record on the Windows Server within the /etc/hosts file, that points from the FQDN to the local IP address, then set the ID server parameter to the FQDN
- Create a DNS record on the Windows Server that points from the FQDN to the public IP address, then set the ID server parameter to the FQDN
- Create a DNS record on the Windows Server within the /etc/hosts file, that points from the FQDN to the public IP address, then set the ID server parameter to the FQDN

The RustDesk application on the server is (in most cases when I did the named things that I tried) showing the "Ready" state. However, everytime I tried to connect to the Windows Server, the same error shows up.

Also interestingly, a connection via the publicly available RustDesk servers works totally fine.

I think it might be a routing issue, but I'm really at the end with my latin. So I'd appreciate any help you'd might have.

**Comment by Cynyster** [comment]
No Solution to this yet?
I have the very same thing happening.  I have the appropriate ports forwarded through the firewall.
(in my case it is a Docker image hosted on a Synology NAS)
Remote session  function normally if the host and client are on the same network. Including those VPN connected.
Rust Desk even shows everything is normal on the client when outside the LAN... right until I try to connect to a host.
Then I get an error saying Failed to connect to relay server.
(Unlike the above post.  I am using IP addresses... no DNS or DHCP involved)

**Comment by mydliar** [comment]
Exactly same setup as @Cynyster, same problem. What are we missing?

**Comment by rustdesk** [comment, marked as answer]
#763 
https://github.com/rustdesk/rustdesk/wiki/FAQ#deploy-rustdesk-server-in-intranet
